### PR TITLE
cli: add system command and subcmds to interact with system API.

### LINKED ISSUE #6747: System API commands for cli
[FEATURE REQUEST]

It would be good to be able to trigger garbage collection in a cluster manually using a command in the nomad binary. There are only two "system" api endpoints which would be convenient to have handy in the cli tool to avoid the need for curl or using a ui.

Suggested command could be something like:

`nomad system garbage-collect` 
`nomad system collect` (or "clean"?)
`nomad system gc`

`nomad system reconcile-summaries`
`nomad system reconcile`
`nomad system rs`

I guess some though needs to be put into command names that are short but descriptive, the above not being the greatest examples. In any case I imagine you would end up with a new top level command "system" with two subcommands.

### PR DESCRIPTION
**Pull request by jrasell**
The system command includes gc and reconcile-summaries subcommands
which covers all currently available system API calls. The help
information is largely pulled from the current Nomad website API
documentation.

closes #6747

### CODE REVIEW

**Comment by angrycub** [comment]
Should `nomad system reconcile-summaries` be `nomad system reconcile summaries` to match the API, allowing for the possibility of `system/reconcile/next-thing-to-reconcile`? 

Command docs would be incredible too.

**Comment by tgross** [comment]
@jrasell the new tests are failing with connection errors:

```
    testlog.go:34: 2020-01-10T13:30:56.472Z [DEBUG] client.server_mgr: new server list: new_servers=[127.0.0.1:9526] old_servers=[127.0.0.1:9526, 127.0.0.1:9526]
    system_reconcile_summaries_test.go:25: expected exit 0, got: 1; Error running system summary reconciliation: Put http://127.0.0.1:4646/v1/system/reconcile/summaries: dial tcp 127.0.0.1:4646: connect: connection refused
```


You'll want to see the address similar to how it's done in [`command/agent_info_test.go#L15-L27`](https://github.com/hashicorp/nomad/blob/4b6f9e800b9f2a712331ecc4bb4e78b2f267249f/command/agent_info_test.go#L15-L27):

```go
func TestAgentInfoCommand_Run(t *testing.T) {
	t.Parallel()
	srv, _, url := testServer(t, false, nil)
	defer srv.Shutdown()

	ui := new(cli.MockUi)
	cmd := &AgentInfoCommand{Meta: Meta{Ui: ui}}

	code := cmd.Run([]string{"-address=" + url})
	if code != 0 {
		t.Fatalf("expected exit 0, got: %d", code)
	}
}
```

**Comment by jrasell** [comment]
@tgross yeah I noticed yesterday; I will try and find time later to figure this out and also make the changes based on feedback from @angrycub

**Comment by endocrimes** [comment]
As an FYI looks like they're failing bc of missing Meta setup to point to the test server https://github.com/hashicorp/nomad/blob/master/command/acl_policy_apply_test.go#L30

After that and @angrycub's suggestions it LGTM

File: website/source/docs/commands/system/gc.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+layout: "docs"
+page_title: "Commands: system gc"
+sidebar_current: "docs-commands-system-gc"
+description: >
+  Run the system garbage collection process.
+---
+
+# Command: system gc
+
+Initializes a garbage collection of jobs, evaluations, allocations, and nodes.
```
Comment (angrycub) [inline website/source/docs/commands/system/gc.html.md.erb]:
```suggestion
Initializes a garbage collection of jobs, evaluations, allocations, and nodes. This is an asynchronous operation.
```

File: website/source/layouts/docs.erb [hunk 2]
```diff
@@ -351,6 +351,17 @@
           <li<%= sidebar_current("docs-commands-status") %>>
             <a href="/docs/commands/status.html">status</a>
           </li>
+          <li<%= sidebar_current("docs-commands-system") %>>
+            <a href="/docs/commands/system.html">system</a>
+            <ul class="nav">
+              <li<%= sidebar_current("docs-commands-system-gc") %>>
+                <a href="/docs/commands/system/gc.html">system gc</a>
```
Comment (angrycub) [inline website/source/layouts/docs.erb]:
```suggestion
                <a href="/docs/commands/system/gc.html">gc</a>
```

File: website/source/layouts/docs.erb [hunk 3]
```diff
@@ -351,6 +351,17 @@
...
+              </li>
+              <li<%= sidebar_current("docs-commands-system-reconcile-summaries") %>>
+                <a href="/docs/commands/system/reconcile-summaries.html">system reconcile summaries</a>
```
Comment (angrycub) [inline website/source/layouts/docs.erb]:
```suggestion
                <a href="/docs/commands/system/reconcile-summaries.html">reconcile summaries</a>
```

File: website/source/docs/commands/system.html.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+layout: "docs"
+page_title: "Commands: system"
+sidebar_current: "docs-commands-system"
+description: >
+  The system command is used to interact with the system API.
+---
+
+# Command: system
+
+The `system` command is used to interact with namespaces.
```
Comment (angrycub) [inline website/source/docs/commands/system.html.md]:
```suggestion
The `system` command is used to interact with the system API. These calls are used for system maintenance and should not be necessary for most users.
```

**Comment by endocrimes** [comment]
:tada:

**Pull request merged**